index.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import d3 from 'd3';
+import nv from 'nvd3';
+
+let SETTINGS = ['x', 'y', 'width', 'height', 'type', 'datum'];
+
+class Chart extends React.Component {
+  static propTypes: {
+    type: React.PropTypes.string.isRequired,
+    configure: React.PropTypes.func
+  }
+
+  componentDidMount() {
+    nv.addGraph(() => {
+      this.chart = nv.models[this.props.type]()
+        .x(this.getValueFunction(this.props.x, 'x'))
+        .y(this.getValueFunction(this.props.y, 'y'))
+        .margin(this.propsByPrefix('margin-') || {})
+        .options(this.getChartOptions());
+
+      !this.props.configure || this.props.configure(this.chart); // hook for configuring the chart
+
+      d3.select(this.refs.svg.getDOMNode()).datum(this.props.datum).call(this.chart);
+
+      nv.utils.windowResize(this.chart.update);
+      return this.chart;
+    });
+  }
+
+  componentDidUpdate() {
+    !this.chart || d3.select(this.refs.svg.getDOMNode()).datum(this.props.datum).call(this.chart);
```

Comment:
I remember that sometimes, with some specific options (I don't remember which ones), this kind of update doesn't reflect changes properly. Said that, I will apply this update and if some bug come up then we could report it to the NVD3 guys.